src/txdb.cpp

```diff
@@ -44,10 +45,33 @@ uint256 CCoinsViewDB::GetBestBlock() const {
     return hashBestChain;
 }
 
+std::vector<uint256> CCoinsViewDB::GetHeadBlocks() const {
+    std::vector<uint256> vhashHeadBlocks;
+    if (!db.Read(DB_HEAD_BLOCKS, vhashHeadBlocks))
+        return std::vector<uint256>();
+    return vhashHeadBlocks;
+}
+
 bool CCoinsViewDB::BatchWrite(CCoinsMap &mapCoins, const uint256 &hashBlock) {
     CDBBatch batch(db);
     size_t count = 0;
     size_t changed = 0;
+    size_t batch_size = (size_t)GetArg("-dbbatchsize", nDefaultDbBatchSize) << 20;
+    if (!hashBlock.IsNull()) {
+        // Read existing heads from database.
+        uint256 tip = GetBestBlock();
+        std::vector<uint256> heads = GetHeadBlocks();
+        // Construct a set with all existing heads, excluding the new tip.
+        std::set<uint256> setHeads(heads.begin(), heads.end());
+        if (setHeads.empty() || !tip.IsNull()) setHeads.insert(tip);
```

Comment:
I am surprised, I always thought the coinbase of the genesis block was not added in the database. This would mean that having it to `null` to say `before processing the genesis`, is the same than enforcing the best block to be at least the genesis block.